src/module/item/base/document.ts

```diff
@@ -285,6 +286,8 @@ class ItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | null> extends Item
                 grant.onDelete ??= "detach";
             }
         }
+
```

Comment:
I don't know that this can safely be set this early.

Reply:
I did a quick test, and it seems that actor.items is filled out pretty early. While the items themselves are incomplete, its the same reference allegedly so it works out.

The only other alternative construction context stuff and storing a private granted by param, but having an additional fake-private property seemed odd.

Reply:
That said, on second glance, it could be that the only reason its safe is because of your prepareData() override that avoids preparing child objects until the parent itself is also initialized. If you'd rather not trust that continues to exist in the future, I could find another way.

Reply:
Prepare sibling data?